Cargo.lock

```diff
@@ -2910,6 +2930,18 @@ dependencies = [
  "serde",
 ]
 
+[[package]]
+name = "op-alloy-rpc-types-engine"
+version = "0.2.11"
+source = "git+https://github.com/alloy-rs/op-alloy?branch=main#95fdc87239d58079c8db2226779545f10758ba77"
+dependencies = [
+ "alloy-primitives",
+ "alloy-rpc-types-engine",
+ "alloy-serde",
+ "op-alloy-protocol",
+ "serde",
+]
```

Comment:
Unfortunately, this is why the build on cannon is failing. `alloy-serde` is not currently hidden behind the `serde` feature flag so it leaks serde and std features into the `no_std` environment.